### ISSUE TITLE
Multiple vms gets the same ip within the same network

### ISSUE THREAD

**Issue by Eslam-Nawara** [threefoldtech/terraform-provider-grid#781]
## Description

- I am trying to deploy a nomad cluster but deployment results in multiple vms with the same ip.

- I've attempt to set the ip of "server1" to some value "10.1.2.2"

### Expected Behavior

- every vm should have different ip with only server1 with ip "10.1.2.2"

![image](https://github.com/threefoldtech/terraform-provider-grid/assets/67752395/afc2d6d0-ea49-4a90-82b2-01a12b1c5705)

### Actual Behavior

- this resulted in more than one vm with ip "10.1.2.2" 

![image](https://github.com/threefoldtech/terraform-provider-grid/assets/67752395/e249afd6-fa99-4870-9007-77d81fcbfec7)

> all of the 3 vms are in the same node and I've tried multible nodes with the same result

**Comment by AbdelrahmanElawady** [comment]
can you share the tf file please

**Comment by Eslam-Nawara** [comment]
https://github.com/threefoldtech/terraform-provider-grid/blob/development_create_nomad_module/examples/resources/nomad_with_modules/main.tf

**Comment by AbdelrahmanElawady** [comment]
It seems a problem with the Go grid client that it assigns the same IP for different deployments when on the same network
https://github.com/threefoldtech/tfgrid-sdk-go/issues/457

**Comment by rawdaGastan** [comment]
### Verification

- Verified using version 1.9.3-rc2. Will be available in the next release `1.9.3`

- Tried the single node example (https://github.com/threefoldtech/terraform-provider-grid/tree/development/examples/resources/singlenode)

- It creates 2 VMs in a deployment and both have different IPs
```console
node1_zmachine1_ip = "172.20.3.2"
node2_zmachine1_ip = "172.20.3.3"
```

### Problem

Tried to deploy 2 deployments of 2 VMs on different nodes, but the IPs were the same:
```console
node1_zmachine1_ip = "172.20.3.2"
node2_zmachine1_ip = "172.20.3.2"
```

**Comment by rawdaGastan** [comment]
### Known issue

Tried to deploy 2 deployments of 2 VMs on different nodes, but the IPs were the same ❌ 
```console
node1_zmachine1_ip = "172.20.3.2"
node2_zmachine1_ip = "172.20.3.2"
```

This is because the parallelism that terraform supports for the resource, so if we deployed 2 or more vm resources it won't wait for the state to be updated, hence the same IP will be assigned for them
